akka-http-core/src/main/scala/akka/http/scaladsl/model/Uri.scala

```diff
@@ -30,6 +30,14 @@ sealed abstract case class Uri(scheme: String, authority: Authority, path: Path,
   def isEmpty: Boolean
 
   /**
+   * Parses the given String into a Query instance.
+   */
+  def query(charset: Charset = UTF8, mode: Uri.ParsingMode = Uri.ParsingMode.Relaxed): Query = queryString match {
```

Comment:
IMHO neither the charset nor the mode is a property of the URI.
Rather these are parameters to the (de)serialization process.
As such I think they should stay as parameters to the parse/render methods.